Resolved issue #445 by changing MSW.cwt's extend length method

wkumler opened the pull request:
Per @jorainer's recommendation, this pull request resolves issue #445 where peaks were dropped from the peaklist during wavelet transformation because the minimum peakwidth was too low. The new behavior also ensures that all wavelet scales passed to the MSW.cwt function are actually run, rather than being selected via weird interaction between the minimum peakwidth and the number of scans in the to-be-transformed chromatogram.

jorainer commented on R/cwTools.R:
```diff
@@ -22,7 +22,13 @@ MSW.cwt <- function (ms, scales = 1, wavelet = "mexh")
         stop("Unsupported wavelet!")
     }
     oldLen <- length(ms)
-    ms <- MSW.extendNBase(ms, nLevel = NULL, base = 2)
```
instead of removing the old with your new code I would suggest to have here something like a:
```r
if (extendLengthMSW) {
    your new code
} else
old code
```

wkumler commented:
Hi @jorainer, sorry for the delay! I've added two commits that enable extendLengthMSW as a parameter within the CentWaveParam that is carried through, preserving the original peakfinding if set to FALSE (the default). I had to append a couple additional files to get it to compile correctly, so it's probably worth taking an extra look at the second commit to make sure it's all still up to scratch.

wkumler commented:
@jorainer Done and done! I ended up manually editing the DESCRIPTION because I couldn't find the roxygen source for it - is that auto-generated?

wkumler commented:
No worries! This is my first time contributing to a large package like this, so your willingness to work with me means a lot. I believe I've updated my fork to the current master, added some documentation and ran `devtools::document()` to generate the `.Rd` files, and updated the unit tests to reflect the changed code. I still got some errors when running `devtools::check()`, mostly relating to BiocParallel not playing nicely with 32-bit R on Windows (namespace functions not found because they aren't passed to the SNOW workers explicitly). But `devtools::test()` now runs without any errors and it compiles completely on my end.
